modules/auxiliary/scanner/http/cisco_secure_desktop.rb

```diff
@@ -0,0 +1,337 @@
+      return false
+    end
+
+    vprint_good("Application appears to allow password logins. Module will continue.")
+
+    groups = Set.new
+    if datastore['GROUP'].empty?
+      vprint_status("Attempt to Enumerate VPN Groups...")
+      groups = enumerate_vpn_groups
+
+      if groups.empty?
+        vprint_error("Unable to enumerate groups")
+      else
+        vprint_good("Enumerated VPN Groups: #{groups.to_a.join(", ")}")
+      end
+
+    else
+      groups << datastore['GROUP']
+    end
+    groups << ""
+
+    vprint_status("Starting login brute force...")
+    groups.each do |group|
+      each_user_pass do |user, pass|
+        do_login(user, pass, group)
+      end
+    end
+  end
+
+  $sdesktop_cookie = nil
+
+  # Verify whether the connection is working or not
+  def check_conn?
+    begin
+      res = send_request_raw('uri' => '/', 'method' => 'GET')
+      if res
+        vprint_good("Server is responsive...")
+        return true
+      end
+    rescue ::Rex::ConnectionRefused,
+           ::Rex::HostUnreachable,
+           ::Rex::ConnectionTimeout,
+           ::Rex::ConnectionError,
+           ::Errno::EPIPE
+    end
+    false
+  end
+
+  def enumerate_vpn_groups
+    headers = {
+      'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+      'Accept-Language' => 'en-US,en;q=0.5',
+      'Cookie' => "webvpnlogin=1; webvpnLang=en; sdesktop=#{$sdesktop_cookie};",
+     }
+
+    res = send_request_raw(
+            'uri' => '/+CSCOE+/logon.html',
+            'method' => 'GET',
+            'headers' => headers,
+          )
+
+    if res &&
+       res.code == 302
+
+      res = send_request_raw(
+              'uri' => '/+CSCOE+/logon.html?fcadbadd=1',
+              'method' => 'GET',
+              'headers' => headers,
+            )
+    end
+
+    groups = Set.new
+    group_name_regex = /onchange="updateLogonForm\(this.value,{(.*):true}\)">/
+
+    if res &&
+       match = res.body.match(group_name_regex)
+
+      group_string = match[1]
+      groups = group_string.scan(/'([\w\-0-9]+)'/).flatten.to_set
+    end
+
+    return groups
+  end
+
+  # Verify whether we're working with Cisco Secure Desktop or not
+  def is_app_secure_desktop?
+    headers = {
+      'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+      'Accept-Language' => 'en-US,en;q=0.5',
+      'Referer' => "https://#{rhost}:#{rport}/",
+      'Cookie' => 'webvpnlogin=1; webvpnLang=en',
+     }
+
+    res = send_request_raw(
+            'uri' => '/+CSCOE+/logon.html',
+            'method' => 'GET',
+            'headers' => headers,
+          )
+
+    if res &&
+       res.code == 302
+
+      res = send_request_raw(
+              'uri' => '/CACHE/sdesktop/install/start.htm',
+              'method' => 'GET',
+              'headers' => headers,
+            )
+    end
+
+    if res &&
+       res.code == 200 &&
+       res.body.match(/Cisco Secure Desktop/)
```

Comment:
Seeing as the data returned from the `match` isn't used, and a regex isn't required, you can do:

```ruby
res.body.include?('Cisco Secure Desktop')
```

Reply:
Seems to work; will commit the change.